diff --git a/node_modules/gatsby/cache-dir/socketIo.js b/node_modules/gatsby/cache-dir/socketIo.js
index c7089f6..9cd9696 100644
--- a/node_modules/gatsby/cache-dir/socketIo.js
+++ b/node_modules/gatsby/cache-dir/socketIo.js
@@ -75,7 +75,7 @@ export default function socketIo() {
 }
 
 const inFlightGetPageDataPromiseCache = {}
-function getPageData(pathname) {
+function getPageData(pathname, initial = true) {
   pathname = normalizePagePath(pathname)
   if (inFlightGetPageDataPromiseCache[pathname]) {
     return inFlightGetPageDataPromiseCache[pathname]
@@ -85,16 +85,30 @@ function getPageData(pathname) {
         delete inFlightGetPageDataPromiseCache[pathname]
         resolve(pageQueryData[pathname])
       } else {
+
+        let responseWatchdog = setTimeout(() => {
+          console.log(`Didn't get response for "getPageData" websocket request`, { pathname, initial })
+        }, 15000)
+
         const onPageDataCallback = msg => {
           if (
             msg.type === `pageQueryResult` &&
             normalizePagePath(msg.payload.id) === pathname
           ) {
+            clearTimeout(responseWatchdog)
             socket.off(`message`, onPageDataCallback)
             delete inFlightGetPageDataPromiseCache[pathname]
             resolve(pageQueryData[pathname])
+
+            // cause much more (useless) websocket traffic
+            setTimeout(() => {
+              delete pageQueryData[pathname]
+              getPageData(pathname, false)
+            }, 50)
           }
         }
+
+
         socket.on(`message`, onPageDataCallback)
 
         socket.emit(`getDataForPath`, pathname)
